### DISCUSSION TITLE
Asynchronous event-driven feature

### DISCUSSION THREAD

**Discussion by lieshanhe** [phalcon/phalcon#489] [Ideas]
Have you guys thought about reworking the route, cookie, and session components in the next version? I recently finished a Phalcon project where I used [Workerman ](https://github.com/walkor/workerman)to drive a console task. It was really cool and significantly boosted performance. But when I tried using Webman to run a Phalcon MVC application, I found out that sessions and cookies only work in php-fpm mode. I know Phalcon provides interfaces for developers to rewrite these components. However, if Phalcon could support running these components outside of fpm mode or even run independently without needing any web server, it would significantly boost Phalcon's competitiveness.

**Comment by dennis6101990leon** [comment]
Hello,

This is a great point about limitations with sessions and cookies in Phalcon outside of php-fpm. While Phalcon offers customization, built in support for these features beyond fpm would be very beneficial. This [HealthInsuranceMarket](https://www.health-insurancemarket.com) could make Phalcon more competitive by allowing standalone or Workerman-driven applications.

**Comment by Jeckerson** [comment]
There is also official [bridge for Swoole](https://github.com/phalcon/bridge-swoole) which do the same and replaces PHP-FPM and Nginx/Apache.
